Chap_API_Sets_Groups.tex

```diff
@@ -169,63 +169,151 @@ \section{Process Groups}
 
 \begin{constantdesc}
+
 %
-\declareconstitemNEW{PMIX_GROUP_DECLINE}
-Decline an invitation to join a \ac{PMIx} group - provided for readability of user code
+\declareconstitemNEW{PMIX_GROUP_INVITED}
+The process has been invited to join a \ac{PMIx} Group - the identifier of the group and the ID's of other invited (or already joined) members will be included in the notification
+
 %
-\declareconstitemNEW{PMIX_GROUP_ACCEPT}
-Accept an invitation to join a \ac{PMIx} group - provided for readability of user code
+\declareconstitemNEW{PMIX_GROUP_LEFT}
+A process has asynchronously left a \ac{PMIx} Group - the process identifier of the departing process will in included in the notification
+
 %
-\declareconstitemNEW{PMIX_GROUP_CONSTRUCT}
-Construct a group composed of the specified processes - used by a \ac{PMIx} server library to direct host operation
+\declareconstitemNEW{PMIX_GROUP_MEMBER_FAILED}
+A member of a \ac{PMIx} Group has abnormally terminated (i.e., without formally leaving the group prior to termination) - the process identifier of the failed process will be included in the notification
+
 %
-\declareconstitemNEW{PMIX_GROUP_DESTRUCT}
-Destruct the specified group - used by a \ac{PMIx} server library to direct host operation
+\declareconstitemNEW{PMIX_GROUP_INVITE_ACCEPTED}
+A process has accepted an invitation to join a \ac{PMIx} Group - the identifier of the group being joined will be included in the notification
+
 %
-\end{constantdesc}
+\declareconstitemNEW{PMIX_GROUP_INVITE_DECLINED}
+A process has declined an invitation to join a \ac{PMIx} Group - the identifier of the declined group will be included in the notification
+
+%
+\declareconstitemNEW{PMIX_GROUP_INVITE_FAILED}
+An invited process failed or terminated prior to responding to the invitation - the identifier of the failed process will be included in the notification.
 
+%
+\declareconstitemNEW{PMIX_GROUP_MEMBERSHIP_UPDATE}
+The membership of a \ac{PMIx} group has changed - the identifiers of the revised membership will be included in the notification.
+
+%
+\declareconstitemNEW{PMIX_GROUP_CONSTRUCT_ABORT}
+Any participant in a \ac{PMIx} group construct operation that returns \refconst{PMIX_GROUP_CONSTRUCT_ABORT} from the \emph{leader failed} event handler will cause all participants to receive an event notifying them of that status. Similarly, the leader may elect to abort the procedure by either returning this error code from the handler assigned to the \refconst{PMIX_GROUP_INVITE_ACCEPTED} or \refconst{PMIX_GROUP_INVITE_DECLINED} codes, or by generating an event for the abort code. Abort events will be sent to all invited or existing members of the group.
+
+%
+\declareconstitemNEW{PMIX_GROUP_CONSTRUCT_COMPLETE}
+The group construct operation has completed - the final membership will be included in the notification.
+
+%
+\declareconstitemNEW{PMIX_GROUP_LEADER_FAILED}
```

Comment:
Why is the 'leader' feature not an implementation dependent detail?

Reply:
This only defines the event, which has to be in the standard so it is portable - afraid I don't understand your issue.